ui/lib/kubernetes/addon/components/page/roles.js

```diff
@@ -12,12 +11,13 @@ export default class ConfigurePageComponent extends Component {
     return getOwner(this).mountPoint;
   }
 
-  @task
-  @waitFor
-  *onDelete(model) {
+  @action
+  async onDelete(model) {
     try {
-      yield model.destroyRecord();
-      this.args.model.roles.removeObject(model);
+      const message = `Successfully deleted role ${model.name}`;
+      await model.destroyRecord();
+      this.args.roles.removeObject(model);
+      this.flashMessages.success(message);
```

Comment:
Since there is no disabled or loading state for this action there is no real advantage of using a concurrency task here that I can see.